.github/workflows/unit_tests.yml

```diff
@@ -16,42 +16,43 @@ jobs:
         # - each xcode version has own simulator os versions.
         # - so to run tests with the target simulator, we have to find a proper xcode version pre-installed and support the target simulator os version.
         #   also, the xcode version (simulator_xcode_version) and simulator os versions (os) are moving target. We have to change these time to time.
+        # - see "https://github.com/actions/runner-images/blob/main/images/macos/macos-11-Readme.md" for installed macOS, xcode and simulator versions.
         include:
-          - os: 15.2
-            device: "iPhone 11"
+          - os: 16.1
+            device: "iPhone 12"
             scheme: "OptimizelySwiftSDK-iOS"
             test_sdk: "iphonesimulator"
             platform: "iOS Simulator"
             os_type: "iOS"
-            simulator_xcode_version: 13.2
-          - os: 14.4
-            device: "iPhone 8"
+            simulator_xcode_version: 14.1
+          - os: 15.5
+            device: "iPhone 12"
             scheme: "OptimizelySwiftSDK-iOS"
             test_sdk: "iphonesimulator"
             platform: "iOS Simulator"
             os_type: "iOS"
-            simulator_xcode_version: 12.4
-          - os: 13.7
+            simulator_xcode_version: 13.4.1
+          - os: 15.5
```

Comment:
can we test os `14` here? since we're already testing 15 above?

Reply:
github actions are terrible in supporting iOS. 15.5 and 16 are only available to macOS 12, which I need to workaround the hang issue. I'll try to extend the range when they get available.